### PR TITLE
doc/27: Added automatically generated capabilities report

### LINKED ISSUE #27: Automatically generate report for scalar function capabilities
Automatically generate a report for scalar function capabilities

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # PostgreSQL Virtual Schema
 
-An implementation of the [PostgreSQL](https://www.postgresql.org/) dialect for the [Exasol Virtual Schema JDBC][virtual-schema-common-jdbc] project.
-Use this dialect if you want to create a Virtual Schema based on a PostgreSQL database.
+An implementation of the [PostgreSQL](https://www.postgresql.org/) dialect for the [Exasol Virtual Schema JDBC][virtual-schema-common-jdbc] project. Use this dialect if you want to create a Virtual Schema based on a PostgreSQL database.
```
Comment (AnastasiiaSergienko) [inline README.md:3]: Let's unify this paragraph with other VS, you can take a template from here, for example 
https://github.com/exasol/mysql-virtual-schema

File: README.md [hunk 2]
```diff
@@ -14,12 +13,13 @@ If you want to set up a Virtual Schema for a different database system, please h
 ### Information for Users
 
 * [PostgreSQL dialect](doc/dialects/postgresql.md)
-* [User Guide][user-guide]
+* [Generic Virtual Schema User Guide][user-guide]
```
Comment (AnastasiiaSergienko) [inline README.md:16]: Don't say Generic here. We have the Generic dialect, it sounds misleading.

File: README.md [hunk 3]
```diff
@@ -14,12 +13,13 @@ If you want to set up a Virtual Schema for a different database system, please h
...
+* [List of supported capabilities](doc/generated/capabilities.md)
```
Comment (AnastasiiaSergienko) [inline README.md:17]: I would move it to information for developers

File: README.md [hunk 4]
```diff
@@ -60,22 +63,42 @@ Running the Virtual Schema requires a Java Runtime version 11 or later.
 | [Project Keeper Maven Plugin][project-keeper-maven-plugin]         | Checking project structure                             | MIT License                   |
 
 [maven-surefire-plugin]: https://maven.apache.org/surefire/maven-surefire-plugin/
+
```
Comment (AnastasiiaSergienko) [inline README.md:66]: I know that intellij does it automatically now, but please remove the empty lines.

File: src/test/java/com/exasol/adapter/dialects/postgresql/docgeneration/CapabilitiesReport.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.Set;
+
+import com.exasol.adapter.capabilities.*;
+import com.exasol.adapter.dialects.SqlDialect;
+import com.exasol.adapter.dialects.SqlDialectFactory;
+import com.exasol.autogeneratedresourceverifier.AutogeneratedResource;
+
+import net.steppschuh.markdowngenerator.table.Table;
+
+public class CapabilitiesReport implements AutogeneratedResource {
```
Comment (AnastasiiaSergienko) [inline src/test/java/com/exasol/adapter/dialects/postgresql/docgeneration/CapabilitiesReport.java:13]: This looks like a generic class. Should it go to common-jdbc? We don't want to copy it for each dialect

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (jakobbraun) [reply README.md:66]: opened an issue: https://youtrack.jetbrains.com/issue/IDEA-263256
Until it's fixed I added `<!-- @formatter:off -->` to disable the formatting for that section.

File: README.md (hunk 3 above)
Comment (jakobbraun) [reply README.md:17]: I think it's important for uses. It's important information for understanding query performance

File: src/test/java/com/exasol/adapter/dialects/postgresql/docgeneration/CapabilitiesReport.java (hunk 5 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/dialects/postgresql/docgeneration/CapabilitiesReport.java:13]: yes. But in a separate PR.

Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/dialects/postgresql/docgeneration/CapabilitiesReport.java:13]: I created an issue: https://github.com/exasol/postgresql-virtual-schema/issues/31

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**